### ISSUE TITLE
"Rebuild Project" is broken in C API samples

### ISSUE THREAD

**Issue by chrisvarns** [google-ar/arcore-android-sdk#149]
Hi,

Attempting to `Build -> Rebuild Project` the `hello_ar_c` sample in Android studio produces the following gradle error:

```
:app:externalNativeBuildDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:externalNativeBuildDebug'.
> Build command failed.
  Error while executing process C:\Users\Varnz\AppData\Local\Android\Sdk\cmake\3.6.4111459\bin\cmake.exe with arguments {--build C:\work\arcore-android-sdk-preview2\samples\hello_ar_c\app\.externalNativeBuild\cmake\debug\arm64-v8a --target hello_ar_native}
  ninja: error: '../../../../build/arcore-native/jni/arm64-v8a/libarcore_sdk.so', needed by '../../../../build/intermediates/cmake/debug/obj/arm64-v8a/libhello_ar_native.so', missing and no known rule to make it
```

After some investigation, the issue here is that the `hello_ar_c/app/build/arcore-native` folder is created by the `extractNativeLibraries` task, which during a normal build gets created before `externalNativeBuild` runs. During a rebuild however, this folder gets removed by a clean step, and so is not present when the `externalNativeBuild` step runs. My gradle-fu is not sufficient to debug exactly which task is cleaning the folder.

**Comment by inio** [comment]
Thanks for the report.  We're looking into it (gradle isn't my specialty either).

**Comment by claywilkinson** [comment]
It looks like when rebuilding, it  is running the configuration phase (which is where the .so files are unpacked), then cleaning.  To fix the script replace the `extractNativeLibraries` task with this snippet.  It causes the unpacking to happen during the build phase and makes the external building tasks depend on the extractNativeLibraries tasks:
```
// Extracts the shared libraries from aars in the natives configuration.
// This is done so that NDK builds can access these libraries.
task extractNativeLibraries() {
    doFirst {
        configurations.natives.files.each { f ->
            copy {
                from zipTree(f)
                into arcore_libpath
                include "jni/**/*"
            }
        }
    }
}

tasks.whenTaskAdded {
    task-> if (task.name.contains("external") && !task.name.contains("Clean")) {
        task.dependsOn(extractNativeLibraries)
    }
}
```

**Comment by inio** [comment]
This should be fixed with the 1.0 samples release today.
